Sources/MongoSwift/APM.swift

```diff
@@ -14,11 +14,247 @@ public struct ConnectionId {
+public enum ServerType: String {
+    case standalone = "Standalone"
+    case mongos = "Mongos"
+    case possiblePrimary = "PossiblePrimary"
+    case rsPrimary = "RSPrimary"
+    case rsSecondary = "RSSecondary"
+    case rsArbiter = "RSArbiter"
+    case rsOther = "RSOther"
+    case rsGhost = "RSGhost"
+    case unknown = "Unknown"
+}
+
+/// A struct describing a mongod or mongos process.
+public struct ServerDescription {
+    /// The hostname or IP and the port number that the client connects to. Note that this is not the
+    /// server's ismaster.me field, in the case that the server reports an address different from the 
+    /// address the client uses.
+    let connectionId: ConnectionId
+
+    /// The last error related to this server.
+    let error: MongoError? = nil // currently we will never set this
+
+    /// The duration of the server's last ismaster call.
+    var roundTripTime: Int64?
+
+    /// The "lastWriteDate" from the server's most recent ismaster response.
+    var lastWriteDate: Date?
+
+    /// The last opTime reported by the server. Only mongos and shard servers 
+    /// record this field when monitoring config servers as replica sets.
+    var opTime: ObjectId?
+
+    /// The type of this server.
+    var type: ServerType = .unknown
+
+    /// The wire protocol version range supported by the server.
+    var minWireVersion: Int32 = 0
+    var maxWireVersion: Int32 = 0
+
+    /// The hostname or IP and the port number that this server was configured with in the replica set.
+    var me: ConnectionId?
+
+    /// Hosts, arbiters, passives: sets of addresses. This server's opinion of the replica set's members, if any.
+    var hosts: [ConnectionId] = []
+    var arbiters: [ConnectionId] = []
+    /// "Passives" are priority-zero replica set members that cannot become primary. 
+    /// The client treats them precisely the same as other members.
+    var passives: [ConnectionId] = []
+
+    /// Tags for this server.
+    var tags: [String: String] = [:]
+
+    /// The replica set name.
+    var setName: String?
+
+    /// The replica set version.
+    var setVersion: Int64?
+
+    /// The election ID where this server was elected, if this is a replica set member that believes it is primary.
+    var electionId: ObjectId?
+
+    /// This server's opinion of who the primary is. 
+    var primary: ConnectionId?
+
+    /// When this server was last checked.
+    let lastUpdateTime: Date? = nil // currently, this will never be set
+
+    /// The logicalSessionTimeoutMinutes value for this server.
+    var logicalSessionTimeoutMinutes: Int64?
+
+    /// An internal initializer to create a `ServerDescription` with just a ConnectionId.
+    internal init(connectionId: ConnectionId) {
+        self.connectionId = connectionId
+    }
+
+    /// An internal function to handle parsing isMaster and setting ServerDescription attributes appropriately.
+    internal mutating func parseIsMaster(_ isMaster: Document) {
```

Comment:
note: I broke this out into its own function because if we eventually choose to implement all the SDAM monitoring spec tests we'd need to write a separate `ServerDescription` init that directly takes in a `ConnectionId` and an isMaster `Document`